centralstore/redis_store.go

```diff
@@ -325,19 +325,52 @@ func (r *RedisBasicStore) GetStatusForTraces(ctx context.Context, traceIDs []str
 	}
 
 	validStates := make(map[CentralTraceState]struct{}, len(statesToCheck))
+	var decisionMade bool
 	for _, state := range statesToCheck {
 		validStates[state] = struct{}{}
+		if (state == DecisionKeep || state == DecisionDrop) && !decisionMade {
+			decisionMade = true
+		}
 	}
 
 	statuses := make([]*CentralTraceStatus, 0, len(statusMapFromRedis))
 	for _, status := range statusMapFromRedis {
-		// only include statuses that are in the statesToCheck list
+		// only include statuses that are in the statesToCheck list with one
+		// exception: if a trace decision is made during stress relief, we need to
+		// find the trace state from the decision cache instead of redis
```

Comment:
```suggestion
		// only include statuses that are in the statesToCheck list.
		// exception: if a trace decision was made during stress relief, we need to
		// find the trace state from the decision cache instead of redis
```